### ISSUE TITLE
INTERNAL COMPILER ERROR with jscomp_error=* 

### ISSUE THREAD

**Issue by l1bbcsg** [google/closure-compiler#3453]
As mentioned in #3451 I encountered this error when trying to compile a large project with `--jscomp_error=*`

The project is private and I cannot share its codebase and not sure how to extract a minimal repro code, compiling just the affected file does not raise any errors.
Apologises this doesn't make a proper report, if you can direct me I'll be happy to expand on it.

Compiler version: `v20190729`

Full list of options: 
```
{ 
  compilation_level: 'ADVANCED_OPTIMIZATIONS',
  language_in: 'ECMASCRIPT_2018',
  language_out: 'ES5',
  warning_level: 'VERBOSE',
  charset: 'UTF8',
  module_resolution: 'BROWSER_WITH_TRANSFORMED_PREFIXES',
  browser_resolver_prefix_replacements: [ 
    'foo/=/absolute/path/to/foo/',
    /* about a dozen more */
  ],
  dependency_mode: 'STRICT',
  entry_point: 'foo.js',
  summary_detail_level: '3',
  externs: [],
  js: [ /* About 100 files */ ],
  checks_only: true,
  jscomp_error: '*'
}
```

Stack trace: 

```
java.lang.RuntimeException: INTERNAL COMPILER ERROR.
Please report this problem.

null
Node(EXPORT): file.js:4:0
export default {
Parent(MODULE_BODY): file.js:4:0
export default {

at com.google.common.base.Preconditions.checkState(Preconditions.java:491)
at com.google.javascript.jscomp.lint.CheckNoMutatedEs6Exports.visitExport(CheckNoMutatedEs6Exports.java:117)
at com.google.javascript.jscomp.lint.CheckNoMutatedEs6Exports.shouldTraverse(CheckNoMutatedEs6Exports.java:85)
at com.google.javascript.jscomp.CombinedCompilerPass$CallbackWrapper.shouldTraverseIfActive(CombinedCompilerPass.java:130)
at com.google.javascript.jscomp.CombinedCompilerPass.shouldTraverse(CombinedCompilerPass.java:169)
at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:851)
at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:979)
at com.google.javascript.jscomp.NodeTraversal.handleModule(NodeTraversal.java:829)
at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:845)
at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:979)
at com.google.javascript.jscomp.NodeTraversal.handleScript(NodeTraversal.java:804)
at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:839)
at com.google.javascript.jscomp.NodeTraversal.traverseChildren(NodeTraversal.java:979)
at com.google.javascript.jscomp.NodeTraversal.traverseBranch(NodeTraversal.java:863)
at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:371)
at com.google.javascript.jscomp.NodeTraversal.traverse(NodeTraversal.java:381)
at com.google.javascript.jscomp.CombinedCompilerPass.process(CombinedCompilerPass.java:154)
at com.google.javascript.jscomp.PhaseOptimizer$NamedPass.process(PhaseOptimizer.java:326)
at com.google.javascript.jscomp.PhaseOptimizer.process(PhaseOptimizer.java:235)
at com.google.javascript.jscomp.Compiler.check(Compiler.java:1022)
at com.google.javascript.jscomp.Compiler.performChecksAndTranspilation(Compiler.java:822)
at com.google.javascript.jscomp.Compiler.lambda$stage1Passes$0(Compiler.java:752)
at com.google.javascript.jscomp.CompilerExecutor$2.call(CompilerExecutor.java:102)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException
... 27 more
```

file.js only exports a JSON object, not terribly large or deep, line 4 is the export declaration:

```
/**
 * @dict
 */
export default {
    // ...
};
```

Without `jscomp_error` the project compiles fine with no warnings.

**Comment by lauraharker** [comment]
Created Google internal issue http://b/139484210

I can reproduce this in the debugger with CHECK_LINT enabled:

https://closure-compiler-debugger.appspot.com/#input0%3Dexport%2520default%2520%257B%250A%2520%2520name%253A%2520'Nombre'%250A%257D%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26CHECK_TYPES%3Dtrue%26CHECK_LINT%3Dtrue%26CLOSURE_PASS%3Dtrue%26PRESERVE_TYPE_ANNOTATIONS%3Dtrue%26PRETTY_PRINT%3Dtrue

One of the lint passes is not handling `export default <some expression>`.
